config/initializers/seeds.rb

```diff
@@ -0,0 +1,51 @@
+return unless (DatabaseSeed.table_exists? rescue(false)) && !Foreman.in_rake?('db:rollback')
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    foreman_seeds = Dir.glob(Rails.root + 'db/seeds.d/*.rb')
+
+    Foreman::Plugin.registered_plugins.each do |name, plugin|
+      begin
+        engine = (name.to_s.tr('-', '_').camelize + '::Engine').constantize
+        foreman_seeds += Dir.glob(engine.root + 'db/seeds.d/*.rb')
+      rescue NameError => e
+        Foreman::Logging.exception("Failed to register plugin #{name}", e)
+      end
+    end
+
+    foreman_seeds = foreman_seeds.sort do |a, b|
+      a.split('/').last <=> b.split('/').last
+    end
+
+    hashes = foreman_seeds.collect { |seed| Digest::SHA256.file(seed).base64digest }
+    new_hash = Digest::SHA256.base64digest(hashes.join)
+
+    db_seed = DatabaseSeed.find_by_id(1)
```

Comment:
I might be nitpicking here, but could this also be a (hidden) setting?

Reply:
... that would make this easier. If you would save the filenames with the individual hashes, you could speed this up a little. And we’d probably have a good example for over engineered code.

Reply:
I need a bit more help understanding here. Hidden setting? Is that simply avoiding a new table to use the existing Settings table to store the hash and keep it from the user? If I went that route, how would I store every individual file and hash? As a JSON blob?

On my devel setup (with Katello and foreman-tasks), the hash calculation and database lookup take 0.013356433 seconds.

Reply:
If you just store a single hash, a setting might be a cheap way to store this data and avoids the extra table. If you want to store an individual hash per file, the extra model is the better fit imho.
Optimizing this for speed would be easy to do, but it’s probably unnecessary.

Reply:
I'm not opposed to that route. @tbrisker @ekohl  your thoughts when you have a chance

Reply:
As always, @timogoebel has a great idea. Do we have a way to hide settings in the UI?

Reply:
is it be worth the effort to save each file's hash? since some files import others and seeding should be idempotent, we'd want to rerun seeds anyways if the combined hash changes. 0.01 seconds at startup to check if we need to run the seeds doesn't sound like much.